crates/orchestrator/src/database/mongodb/mod.rs

```diff
@@ -101,25 +112,40 @@ impl Database for MongoDb {
 
     #[tracing::instrument(skip(self), fields(function_type = "db_call"), ret, err)]
     async fn get_job_by_id(&self, id: Uuid) -> Result<Option<JobItem>> {
+        let start = Instant::now();
         let filter = doc! {
             "id":  id
         };
         tracing::debug!(job_id = %id, category = "db_call", "Fetched job by ID");
+        let attributes = [
+            KeyValue::new("db_operation_name", "get_job_by_id"),
+            KeyValue::new("db_operation_id", format!("{:?}", id)),
+        ];
+        let duration = start.elapsed();
+        ORCHESTRATOR_METRICS.db_calls_response_time.record(duration.as_secs_f64(), &attributes);
         Ok(self.get_job_collection().find_one(filter, None).await?)
     }
 
     #[tracing::instrument(skip(self), fields(function_type = "db_call"), ret, err)]
     async fn get_job_by_internal_id_and_type(&self, internal_id: &str, job_type: &JobType) -> Result<Option<JobItem>> {
+        let start = Instant::now();
         let filter = doc! {
             "internal_id": internal_id,
             "job_type": mongodb::bson::to_bson(&job_type)?,
         };
         tracing::debug!(internal_id = %internal_id, job_type = ?job_type, category = "db_call", "Fetched job by internal ID and type");
+        let attributes = [
+            KeyValue::new("db_operation_name", "get_job_by_internal_id_and_type"),
+            KeyValue::new("db_operation_id", format!("{:?}", internal_id)),
+        ];
+        let duration = start.elapsed();
+        ORCHESTRATOR_METRICS.db_calls_response_time.record(duration.as_secs_f64(), &attributes);
```

Comment:
can we make this a macro maybe?

Reply:
Not implemented as per offline discussion